src/NuGet.Core/NuGet.Protocol/Resources/OwnerDetailsUriTemplateResourceV3.cs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using NuGet.Protocol.Core.Types;
+
+namespace NuGet.Protocol.Resources
+{
+    /// <summary>Owner Details Uri Template for NuGet V3 HTTP feeds.</summary>
+    /// <remarks>Not intended to be created directly. Use <see cref="SourceRepository.GetResourceAsync{T}(CancellationToken)"/>
+    /// with <see cref="OwnerDetailsUriTemplateResourceV3"/> for T, and typecast to this class.
+    public class OwnerDetailsUriTemplateResourceV3 : INuGetResource
+    {
+        private readonly string _template;
+
+        private OwnerDetailsUriTemplateResourceV3(string template)
+        {
+            _template = template ?? throw new ArgumentNullException(nameof(template));
+        }
+
+        /// <summary>
+        /// Creates the specified Owner Details Uri template provided by the server if it exists and is valid.
+        /// </summary>
+        /// <param name="uriTemplate">The Absolute Uri template provided by the server.</param>
+        /// <returns>A valid Owner Details Uri template, or null.</returns>
+        public static OwnerDetailsUriTemplateResourceV3? CreateOrNull(Uri? uriTemplate)
+        {
+            if (uriTemplate == null || uriTemplate.OriginalString.Length == 0 || !uriTemplate.IsAbsoluteUri)
+            {
+                return null;
+            }
+
+            string absoluteUri = uriTemplate.OriginalString;
+            if (string.IsNullOrWhiteSpace(absoluteUri)
+                || !IsValidUriTemplate(absoluteUri))
+            {
+                return null;
+            }
+
+            return new OwnerDetailsUriTemplateResourceV3(absoluteUri);
+        }
+
+        private static bool IsValidUriTemplate(string absoluteUri)
+        {
+            Uri? uri;
+            var isValidUri = Uri.TryCreate(absoluteUri, UriKind.Absolute, out uri);
```

Comment:
I noticed that method as well, but unfortunately, it returns `false` because of the braces in the URI template provided by the server resource, as shown below.

My suspicion is this is why the [PackageDetailsUriResourceV3](https://github.com/NuGet/NuGet.Client/blob/784274cc9b2002ca73055e8134064261118d8a01/src/NuGet.Core/NuGet.Protocol/Resources/PackageDetailsUriResourceV3.cs#L19) does this custom validation, and it's why I copied that for the `OwnerDetailsUriTemplateResourceV3` as well. 

If you have other suggestions, let me know. I'm a little hesitant to modify the `PackageDetailsUriResourceV3` in this PR, but could consider refactoring it as well separately.

![image](https://github.com/NuGet/NuGet.Client/assets/49205731/589e80b5-865f-42e1-8994-944b6718f206)

Reply:
`CreateOrNull` method in [PackageDetailsUriResourceV3](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Protocol/Resources/PackageDetailsUriResourceV3.cs#L19) takes a `string` as parameter but in this resource the same method has `Uri` as parameter. Is this could be a reason why we are converting the Uri to string and then string to Uri for validation purposes?

Reply:
I've reduced some of the validation logic there, have a look and see if that addresses your concern.